### DISCUSSION TITLE
Enabling a 'save' button based on all fields in form being valid

### DISCUSSION THREAD

**Discussion by chriswhiteoco** [danvick/flutter_form_builder#1322] [Q&A]
I need to enable a save button when all the fields in a form are valid.

I tried to find an example of this but couldn't. 

Any suggestions or pointers to some example?

**Comment by deandreamatias** [comment]
You need use _formKey.currentState?.saveAndValidate() to validate the form and with this result, enable or disable the button.
Can implement this logic into a Viewmodel.

~~Currently we don't have an example with this exactly behavior~~

**Comment by deandreamatias** [comment]
Update 2026: Now exist the example https://github.com/flutter-form-builder-ecosystem/flutter_form_builder/pull/1508
